lib/facade/MonitoringAspect.ts

```diff
@@ -0,0 +1,55 @@
+import { IConstruct, IAspect } from "monocdk";
+import * as apigw from "monocdk/aws-apigateway";
+import * as apigwv2 from "monocdk/aws-apigatewayv2";
+import * as acm from "monocdk/aws-certificatemanager";
+
+import { MonitoringFacade } from "./MonitoringFacade";
+
+export interface MonitoringAspectProps {
+  /**
+   * Automatically monitor ACM Certificats in the scope
+   */
+  readonly acm?: boolean;
+
+  /**
+   * Automatically monitor API Gateway Rest Apis in the scope
+   * @default true
+   */
+  readonly apiGateway?: boolean;
+
+  /**
+   * Automatically monitor API Gateway HTTP Apis in the scope
+   * @default true
+   */
+  readonly apiGatewayV2?: boolean;
+}
+
+/**
+ * A CDK aspect that adds support for monitoring all resources within scope.
+ */
+export class MonitoringAspect implements IAspect {
+  constructor(
+    private readonly monitoringFacade: MonitoringFacade,
+    private readonly props: MonitoringAspectProps = {}
+  ) {}
+
+  public visit(node: IConstruct): void {
+    const monitorAcmCertificate = this.props.acm ?? true;
+    if (monitorAcmCertificate && node instanceof acm.Certificate) {
+      this.monitoringFacade.monitorCertificate({
+        certificate: node,
+        alarmFriendlyName: node.node.path,
+      });
+    }
```

Comment:
Long-term, we should probably split each to a separate method? What do you think? Or even create some "node type to monitoring" mapping to make it easier?

Reply:
Thinking even bigger:

What about extending each Monitoring with Aspect, so that each monitoring would be able to identify its own construct? And the we would have the MonitorAllAspect, which would just forward the node to all known aspects?

Reply:
Added the 1st suggestion in the latest change.
As per the 2nd one, we would need to see if that is even possible since each monitoring does not have access to the instantiated monitoring facade.